### PR TITLE
Improve quadratic duality in doc

### PR DESCRIPTION
**Pull request by blegat**
Preview at https://jump.dev/MathOptInterface.jl/previews/PR1717/

### CODE REVIEW

**Comment by matbesancon** [comment]
is there a preview for it?

**Comment by blegat** [comment]
> is there a preview for it?

Added in the PR description

**Comment by matbesancon** [comment]
I would add a reference / link / explanation for the normal cone. This is typically not taught in basic optimization classes.

>   For instance, if ``\mathcal{C}_i`` is ``\{ x \in \mathbb{R} : x \le 0 \}``, this means that
  if ``\frac{1}{2}x^TQ_ix + a_i^T x + b_i`` is nonzero at ``x^\star`` then ``y_i^\star = 0``.
This is the classical complementary slackness condition.

For notation, I would not use `x` in the definition of the cone, since it's the affine image of x which is in the cone

**Comment by matbesancon** [comment]
The affinely constrained part ends up being rather big, I would have it as a subsection rather than a note

File: docs/src/background/duality.md [hunk 1]
```diff
@@ -149,35 +192,14 @@ A pair of primal-dual variables $(x^\star, y^\star)$ is optimal if
...
+  ``\{ z \in \mathbb{R} : z \le 0 \}``, this means that if
+  ``\frac{1}{2}x^TQ_ix + a_i^T x + b_i`` is nonzero at ``x^\star`` then
+  ``y_i^\star = 0``. This is the classical complementary slackness condition.
 
 If ``\mathcal{C}_i`` is a vector set, the discussion remains valid with
 ``y_i(\frac{1}{2}x^TQ_ix + a_i^T x + b_i)`` replaced with the scalar product
 between ``y_i`` and the vector of scalar-valued quadratic functions.
-
-!!! note
-    For quadratic programs with only affine constraints, the optimality condition
```
Comment (matbesancon) [inline docs/src/background/duality.md:163]: did you remove the affine constraint section altogether?

Comment (blegat) [reply docs/src/background/duality.md:163]: I moved it up

**Review by matbesancon — APPROVED** [review]

**Pull request merged**